src/TropicalGeometry/groebner_basis.jl

```diff
@@ -0,0 +1,80 @@
+###
+# Computing (tropical) Groebner bases in Oscar
+# ============================================
+#
+# For a definition of tropical Groebner basis see Section 2.4 in:
+#   D. Maclagan, B. Sturmfels: Introduction to tropical geometry
+# To see how they can be computed using standard bases see:
+#   T. Markwig, Y. Ren: Computing tropical varieties over fields with valuation
+###
+
+
+
+#=======
+return true if f is homogeneous (w.r.t. total degree)
+return false otherwise
+=======#
+function is_homogeneous(f::Union{AbstractAlgebra.Generic.MPoly{K},fmpq_mpoly,fmpz_mpoly} where {K})
+  leadexpv,tailexpvs = Iterators.peel(exponent_vectors(f))
+  d = sum(leadexpv)
+  for tailexpv in tailexpvs
+    if d!=sum(tailexpv)
+      return false
+    end
+  end
+  return true
+end
+export is_homogeneous
+
+function is_homogeneous(I::MPolyIdeal{K} where {K})
```

Comment:
I was actually surprised to see that this works (but upon re-reading the docs, it's clearly OK), so I learned something, thanks :-)

That said, I am not sure it is useful to write it like this: If you don't reference `K` in the code, then this can (and should) be replaced by
```suggestion
function is_homogeneous(I::MPolyIdeal)
```
If you do plan to reference `K` in the code, this won't work, you'll have to write
```suggestion
function is_homogeneous(I::MPolyIdeal{K}) where {K}
```